agenta-cli/agenta/sdk/agenta_init.py

```diff
@@ -61,23 +75,33 @@ def init(
                     f"Warning: Your configuration will not be saved permanently since app_name and base_name are not provided."
                 )
             else:
-                app_id = client.get_app_by_name(
-                    app_name=app_name, host=host, api_key=api_key
-                )
-                base_id = client.get_base_by_app_id_and_name(
-                    app_id=app_id, base_name=base_name, host=host, api_key=api_key
-                )
+                try:
+                    get_app_id = client.list_apps_apps_get(app_name=app_name)
+                    app_id = get_app_id.app_id
+
+                    if not app_id:
+                        raise APIRequestError(
+                            f"App with name {app_name} does not exist on the server."
+                        )
+
+                    get_base_id = client.list_bases_bases_get(
+                        app_id=app_id, base_name=base_name
+                    )
+                    base_id = get_base_id.base_id
+                except Exception as ex:
+                    raise APIRequestError(
+                        f"Failed to get base id and/or app_id from the server with error: {ex}"
+                    )
         self.base_id = base_id
         self.host = host
         self.api_key = api_key
-        self.config = Config(base_id=base_id, host=host, api_key=api_key)
+        self.config = Config(base_id=base_id, host=host)
```

Comment:
I think you should pass either the client or the api key here due to the comment above

Reply:
See https://github.com/Agenta-AI/agenta/pull/1041#discussion_r1425782092